tests/components/wmspro/test_cover.py

```diff
@@ -0,0 +1,223 @@
+"""Test the wmspro diagnostics."""
+
+from unittest.mock import AsyncMock, patch
+
+from homeassistant.components.wmspro.const import DOMAIN, MANUFACTURER
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry
+
+
+async def test_cover_device(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_hub_ping: AsyncMock,
+    mock_hub_configuration_prod: AsyncMock,
+    mock_hub_status_prod_awning: AsyncMock,
+) -> None:
+    """Test that a cover device is created correctly."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_prod.mock_calls) == 1
+    assert len(mock_hub_status_prod_awning.mock_calls) == 2
+
+    device_registry = dr.async_get(hass)
+    device_entry = device_registry.async_get_device(identifiers={(DOMAIN, "58717")})
+    assert device_entry is not None
+    assert device_entry.manufacturer == MANUFACTURER
+    assert device_entry.name == "Markise"
+    assert device_entry.serial_number == 58717
+
+
+async def test_cover_update(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_hub_ping: AsyncMock,
+    mock_hub_configuration_prod: AsyncMock,
+    mock_hub_status_prod_awning: AsyncMock,
+) -> None:
+    """Test that a cover entity is created and updated correctly."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    assert len(mock_hub_ping.mock_calls) == 1
+    assert len(mock_hub_configuration_prod.mock_calls) == 1
+    assert len(mock_hub_status_prod_awning.mock_calls) == 2
+
+    entity = hass.states.get("cover.markise")
+    assert entity is not None
+    assert entity.name == "Markise"
+
+    await async_setup_component(hass, "homeassistant", {})
+    await hass.services.async_call(
+        "homeassistant", "update_entity", {"entity_id": entity.entity_id}, blocking=True
+    )
+
+    assert len(mock_hub_status_prod_awning.mock_calls) == 3
```

Comment:
We'd rather repatch the mock to return a different value, then use the freezer to skip the polling interval (5s) and then observe the state machine again and see if the state got updated.

Reply:
This test was not created to test state changes, but to have 100% test coverage for the `async_update` function.

The other tests are testing state changes.